runtime/commonMain/src/kotlinx/serialization/Shorthands.kt

```diff
@@ -26,9 +26,8 @@ public fun SerialDescriptor.elementDescriptors(): List<SerialDescriptor> {
 }
 
 /**
- * Creates a [List] out of a serial names of [this] descriptor's elements [SerialDescriptor.getElementDescriptor].
- *
- * Size of a list is equal to [SerialDescriptor.elementsCount].
+ * Returns a [List] out of all serial names of serial descriptor [elements][SerialDescriptor.getElementDescriptor].
+ * TODO always allocates, also revisit
```

Comment:
I'm not sure `TODO` is a good line for documentation, maybe put it as a separate regular `//` comment?)

Reply:
btw, it can be improved also with `if (this is SerialClassDescImpl) return names` (assuming `names` visibility is raised to internal and we do not care if someone tries to cast the result to an `ArrayList`)